data/pigui/modules/system-view-ui.lua

```diff
@@ -0,0 +1,524 @@
+
+		-- indicators
+		local stackedSize = indicatorSize
+		local stackStep = Vector2(10, 10)
+		if group.hasPlayer then
+			ui.addIcon(mainCoords, icons.square, svColor.PLAYER, stackedSize, ui.anchor.center, ui.anchor.center)
+			stackedSize = stackedSize + stackStep
+		end
+		if group.hasNavTarget then
+			ui.addIcon(mainCoords, icons.square, svColor.NAV_TARGET, stackedSize, ui.anchor.center, ui.anchor.center)
+			stackedSize = stackedSize + stackStep
+		end
+		if group.hasCombatTarget then
+			ui.addIcon(mainCoords, icons.square, svColor.COMBAT_TARGET, stackedSize, ui.anchor.center, ui.anchor.center)
+			stackedSize = stackedSize + stackStep
+		end
+		if mainObject.type == Projectable.OBJECT and mainObject.base == Projectable.PLANNER then ui.addIcon(mainCoords, icons.square, svColor.PLANNER, indicatorSize, ui.anchor.center, ui.anchor.center) end
+
+		ui.addIcon(mainCoords, getBodyIcon(mainObject), getColor(mainObject), iconsize, ui.anchor.center, ui.anchor.center)
+
+		if should_show_label then
+			local label = getLabel(mainObject)
+			if group.objects then
+				label = label .. " (" .. #group.objects .. ")"
+			end
+			ui.addStyledText(mainCoords + Vector2(label_offset,0), ui.anchor.left, ui.anchor.center, label , getColor(mainObject), pionillium.small)
+		end
+		local mp = ui.getMousePos()
+
+		if mainObject.type == Projectable.OBJECT and (mainObject.base == Projectable.SYSTEMBODY or mainObject.base == Projectable.SHIP or mainObject.base == Projectable.PLAYER) then
+			-- mouse release handler for right button
+			if (mp - mainCoords):length() < click_radius then
+				if not ui.isAnyWindowHovered() and ui.isMouseReleased(1) then
+					ui.openPopup("target" .. objectCounter)
+				end
+			end
+			-- make popup
+			ui.popup("target" .. objectCounter, function()
+				local isObject = mainObject.type == Projectable.OBJECT
+				local isSystemBody = isObject and mainObject.base == Projectable.SYSTEMBODY
+				local isShip = isObject and not isSystemBody and mainObject.ref:IsShip()
+				ui.text(getLabel(mainObject))
+				ui.separator()
+				if ui.selectable(lc.CENTER, false, {}) then
+					systemView:SetSelectedObject(mainObject.type, mainObject.base, mainObject.ref)
+				end
+				if (isShip or isSystemBody and mainObject.ref.physicsBody) and ui.selectable(lc.SET_AS_TARGET, false, {}) then
+					if isSystemBody then
+						player:SetNavTarget(mainObject.ref.physicsBody)
+					else 
+						if combatTarget == mainObject.ref then player:SetCombatTarget(nil) end
+						player:SetNavTarget(mainObject.ref)
+					end
+				end
+				if isShip and ui.selectable(lc.SET_AS_COMBAT_TARGET, false, {}) then
+					if navTarget == mainObject.ref then player:SetNavTarget(nil) end
+					player:SetCombatTarget(mainObject.ref)
+				end
+			end)
+		end
+		-- mouse release handler
+		if ui.isMouseReleased(0) then
+			systemView:SetVisibility("MOUSE_UP")
+			Input:DisableViewTransformationMode()
+		end
```

Comment:
I'd *strongly* recommend you move this functionality to SystemView.cpp instead; see PlayerShipController.cpp in #4842 for an example implementation.

Reply:
Done! See d9ac37975b2f4161bc1350600ca22d3fe5cedf98